enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/components/Notebook.tsx

```diff
@@ -0,0 +1,46 @@
+import { useInteractiveQuestionContext } from "embedding-sdk/components/public/InteractiveQuestion/context";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  runQuestionQuery,
+  updateQuestion,
+} from "metabase/query_builder/actions";
+import { default as QBNotebook } from "metabase/query_builder/components/notebook/Notebook";
+import {
+  getIsDirty,
+  getIsResultDirty,
+  getIsRunnable,
+} from "metabase/query_builder/selectors";
+import { getSetting } from "metabase/selectors/settings";
+import type Question from "metabase-lib/v1/Question";
+
+export const Notebook = () => {
+  const { question, setIsNotebookOpen } = useInteractiveQuestionContext();
+
+  const isDirty = useSelector(getIsDirty);
+  const isRunnable = useSelector(getIsRunnable);
+  const isResultDirty = useSelector(getIsResultDirty);
+  const reportTimezone = useSelector(state =>
+    getSetting(state, "report-timezone-long"),
+  );
+
+  const dispatch = useDispatch();
+
+  return (
+    question && (
+      <QBNotebook
+        question={question}
+        isDirty={isDirty}
+        isRunnable={isRunnable}
+        isResultDirty={Boolean(isResultDirty)}
+        reportTimezone={reportTimezone}
+        readOnly={false}
+        updateQuestion={(question: Question) =>
+          dispatch(updateQuestion(question))
+        }
+        runQuestionQuery={() => dispatch(runQuestionQuery())}
+        setQueryBuilderMode={() => setIsNotebookOpen(false)}
```

Comment:
is this like an event handler when the query builder mode is changed? I'm not sure how "setQueryBuilderMode" and "setIsNotebookOpen" relates to one another

Reply:
Yep, the usage in the main app is (`frontend/src/metabase/query_builder/components/notebook/Notebook.tsx`)
```
 async function visualize() {
    // Only cleanup the question if it's dirty, otherwise Metabase
    // will incorrectly display the Save button, even though there are no changes to save.
    if (isDirty) {
      cleanupQuestion();
    }
    // switch mode before running otherwise URL update may cause it to switch back to notebook mode
    await setQueryBuilderMode("view");
    if (isResultDirty) {
      await runQuestionQuery();
    }
  }
```
With the different (mutually-exclusive) logic that the main QB uses, it makes sense but in the SDK it's a little different